core/src/main/scala/caliban/GraphQLResponse.scala

```diff
@@ -42,6 +45,45 @@ private object GraphQLResponseCirce {
 
 }
 
+private object GraphQLResponseJsoniterJson {
+  import com.github.plokhotnyuk.jsoniter_scala.core._
+  val graphQLResponseCodec: JsonValueCodec[GraphQLResponse[Any]] = new JsonValueCodec[GraphQLResponse[Any]] {
+    private val responseValueCodec: JsonValueCodec[ResponseValue] =
+      implicitly[JsonValueCodec[ResponseValue]]
+    private def handleError(out: JsonWriter)(err: Any): Unit =
+      err match {
+        case ce: CalibanError =>
+          out.writeRawVal(writeToString(ce)(ErrorJsoniter.errorValueCodec).getBytes("UTF-8"))
+        case _ =>
+          out.writeObjectStart()
+          out.writeKey("message")
+          out.writeVal(err.toString)
+          out.writeObjectEnd()
+      }
+    def decodeValue(in: JsonReader, default: GraphQLResponse[Any]): GraphQLResponse[Any] =
+      default // TODO decode GraphQLResponse
+    def encodeValue(x: GraphQLResponse[Any], out: JsonWriter): Unit = {
+      out.writeObjectStart()
+      out.writeKey("data")
+      out.writeRawVal(writeToString(x.data)(responseValueCodec).getBytes("UTF-8"))
+      x.extensions.foreach { x1 =>
+        out.writeKey("extensions")
+        out.writeRawVal(writeToString(x1.asInstanceOf[ResponseValue])(responseValueCodec).getBytes("UTF-8"))
```

Comment:
More efficient would be using a codec for values of the collection immediately without redundant conversions though strings and byte arrays:
```
responseValueCodec.encodeValue(x.data, out)
x.extensions.foreach(x1 => responseValueCodec.encodeValue(x1, out))
```

Reply:
Wow! thanks for all the comments. I'll look into them as soon as I find some time.